### PR TITLE
Fix WithCargo desync.

### LINKED ISSUE #9094: Desync when Cargo.InitialUnits is used
Game Crash when i build modded Plane
 on this version: "Civillian Outrage BETA 48a"   http://resource.openra.net/maps/4168
out of sync 

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/25838582-desync-when-cargo-initialunits-is-used?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by pchote**
Adjusting world state from render code is a bad idea.
Closes #9094.

### CODE REVIEW

**Comment by Mailaender** [comment]
This probably solves the problem I had when I adding manual Carryalls to Dune 2000. :+1:

File: OpenRA.Mods.Cnc/Traits/Render/WithCargo.cs [hunk 1]
```diff
@@ -45,6 +45,16 @@ public WithCargo(Actor self, WithCargoInfo info)
 			body = self.Trait<IBodyOrientation>();
 		}
 
+		public void Tick(Actor self)
+		{
+			foreach (var c in cargo.Passengers)
+			{
+				var cargoFacing = c.TraitOrDefault<IFacing>();
+				if (facing != null && cargoFacing != null)
```
Comment (RoosterDragon) [inline OpenRA.Mods.Cnc/Traits/Render/WithCargo.cs]: Can we lift the null check on facing out of the loop?

Comment (Mailaender) [inline OpenRA.Mods.Cnc/Traits/Render/WithCargo.cs]: How? The trait lookup is done on the passengers who are looped upon.

Comment (pchote) [inline OpenRA.Mods.Cnc/Traits/Render/WithCargo.cs]: `facing` is a trait on the transport.

Comment (Mailaender) [inline OpenRA.Mods.Cnc/Traits/Render/WithCargo.cs]: Right. You can actually just skip the whole loop then. Mind to do that simple change so we can release?

**Comment by Mailaender** [comment]
This is taking too long already. Superseded by https://github.com/OpenRA/OpenRA/pull/9172.

**Pull request closed without merging**